controllers/release/release_adapter_test.go

```diff
@@ -0,0 +1,421 @@
+		link, _ := adapter.getTargetReleaseLink()
+		Expect(reflect.TypeOf(link)).
+			To(Equal(reflect.TypeOf(&appstudiov1alpha1.ReleaseLink{})))
+		Expect(link.Spec.Application).Should(ContainSubstring("test-app"))
+		Expect(link.Spec.ReleaseStrategy).Should(ContainSubstring(releaseStrategy.Name))
+	})
+
+	It("can mark the status for a given Release according to the pipelineRun status", func() {
+		pipelineRun := tekton.NewReleasePipelineRun("test-pipeline", testNamespace).AsPipelineRun()
+		pipelineRun.Name = "foo" // PipelineRun needs a name before registering the Release status
+		err := adapter.registerReleaseStatusData(pipelineRun, releaseStrategy)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(adapter.release.Status.StartTime).To(Not(BeNil()))
+		pipelineRun.Status.MarkSucceeded("succeeded", "set it to succeeded")
+		Expect(pipelineRun.IsDone()).To(BeTrue())
+
+		err = adapter.registerReleasePipelineRunStatus(pipelineRun)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(release.HasSucceeded()).To(BeTrue())
+
+		Expect(adapter.release.Status.StartTime).ToNot(BeNil()) // Double check that we still have a valid StartTime
+
+		release.Status.Conditions = []metav1.Condition{} // Clear up previous condition so isDone returns false
+		pipelineRun.Status.MarkFailed("failed", "set it to failed")
+		err = adapter.registerReleasePipelineRunStatus(pipelineRun)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(release.HasSucceeded()).To(BeFalse())
+	})
+
+	It("can add the release information to its status and mark it as Running", func() {
+		// err should be nil if releaseStrategy or pipelineRun is nil
+		err := adapter.registerReleaseStatusData(nil, nil)
+		Expect(err).To(BeNil())
+
+		pipelineRun := tekton.NewReleasePipelineRun("test-pipeline", testNamespace).AsPipelineRun()
+		pipelineRun.Name = "new-name"
+		err = adapter.registerReleaseStatusData(pipelineRun, releaseStrategy)
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(release.Status.ReleasePipelineRun, pipelineRun.GetName())
+		Expect(release.Status.Conditions[0].Status == "Unknown")
+		Expect(release.Status.Conditions[0].Reason == "Running")
+	})
+
+	It("can update the Status of a release", func() {
+		patch := client.MergeFrom(adapter.release.DeepCopy())
+		adapter.release.Status.ReleasePipelineRun = "foo/test"
+
+		Expect(k8sClient.Status().Patch(ctx, release, patch)).Should(Succeed())
+		Expect(release.Status.ReleasePipelineRun).
+			To(Equal("foo/test"))
+	})
+	It("can get an existing ApplicationSnapshot", func() {
+		snapshot, err := adapter.getApplicationSnapshot()
+		Expect(err).Should(Succeed())
+		Expect(reflect.TypeOf(snapshot)).
+			To(Equal(reflect.TypeOf(&appstudioshared.ApplicationSnapshot{})))
+
+		Expect(k8sClient.Delete(ctx, snapshot)).Should(Succeed())
+
+		k8sClient.Get(ctx, types.NamespacedName{
+			Name:      snapshot.Name,
+			Namespace: testNamespace,
+		}, applicationSnapshot)
+
+		Eventually(func() bool {
```

Comment:
It's a nice test actually. It's not done in other `get*` functions, but it could be done. Up to you ;)

Reply:
I will skip this for now.